examples/api/lib/cupertino/segmented_control/cupertino_segmented_control.1.dart

```diff
@@ -0,0 +1,141 @@
+      home: SegmentedControlExample(),
+    );
+  }
+}
+
+class SegmentedControlExample extends StatefulWidget {
+  const SegmentedControlExample({super.key});
+
+  @override
+  State<SegmentedControlExample> createState() => _SegmentedControlExampleState();
+}
+
+class _SegmentedControlExampleState extends State<SegmentedControlExample> {
+  Sky _selectedSegment = Sky.midnight;
+  bool _toggleOne = false;
+  bool _toggleAll = true;
+  Set<Sky> _disabledChildren = <Sky>{};
+
+  @override
+  Widget build(BuildContext context) {
+    return CupertinoPageScaffold(
+      backgroundColor: skyColors[_selectedSegment],
+      navigationBar: CupertinoNavigationBar(
+        // This Cupertino segmented control has the enum "Sky" as the type.
+        middle: CupertinoSegmentedControl<Sky>(
+          disabledChildren: _disabledChildren,
+          selectedColor: skyColors[_selectedSegment],
+          borderColor: CupertinoColors.transparent,
+          pressedColor: CupertinoColors.systemFill,
+          unselectedColor: CupertinoColors.systemGrey4,
+          // Provide horizontal padding around the children.
+          padding: const EdgeInsets.symmetric(horizontal: 12),
+          // This represents a currently selected segmented control.
+          groupValue: _selectedSegment,
+          // Callback that sets the selected segmented control.
+          onValueChanged: (Sky value) {
+            setState(() {
+              _selectedSegment = value;
+            });
+          },
+          children: const <Sky, Widget>{
+            Sky.midnight: Padding(
+              padding: EdgeInsets.symmetric(horizontal: 20),
+              child: Text('Midnight'),
+            ),
+            Sky.viridian: Padding(
+              padding: EdgeInsets.symmetric(horizontal: 20),
+              child: Text('Viridian'),
+            ),
+            Sky.cerulean: Padding(
+              padding: EdgeInsets.symmetric(horizontal: 20),
+              child: Text('Cerulean'),
+            ),
+          },
+        ),
+      ),
+      child: Center(
+        child: Column(
+          mainAxisAlignment: MainAxisAlignment.center,
+          children: <Widget>[
+            Text(
+              'Selected Segment: ${_selectedSegment.name}',
+              style: const TextStyle(color: CupertinoColors.white),
+            ),
+            const SizedBox(height: 20),
+            Row(
+              mainAxisSize: MainAxisSize.min,
+              children: <Widget>[
+                const Text('Disable one segment', style: TextStyle(color: CupertinoColors.white)),
+                CupertinoSwitch(
+                  value: _toggleOne,
+                  onChanged: (bool value) {
+                    setState(() {
+                      _toggleOne = value;
+                      if (value) {
+                        _toggleAll = false;
+                        _disabledChildren = <Sky>{Sky.midnight};
+                      } else {
+                        _toggleAll = true;
```

Comment:
>Why _toggleAll is set to true if _toggleOne is false? This part looks a little confusing. Seems if toggleOne is true, it disables the first segment, but if toggleAll is true, it enables all segments.🤔

In this example, my purpose is to create clear cases, please also watch the demo video in the [original comment](https://github.com/flutter/flutter/pull/152813#issue-2446562771)
1. User can disable one segment (1st segment for eg)
2. User can disable/enable all segments

So, when user toggles one Switch, it will also reset the remaining Switch.

>Since this example looks similar with cupertino_segmented_control.0.dart, maybe we can just add the Switches in that example instead of creating a new one:)

My initial purpose is to create separate examples for disabled segments. But I can do this if it's cleaner :)